src/amo/components/AddonCompatibilityError/index.js

```diff
@@ -14,43 +16,78 @@ import {
   INCOMPATIBLE_UNDER_MIN_VERSION,
   INCOMPATIBLE_UNSUPPORTED_PLATFORM,
 } from 'core/constants';
-import _log from 'core/logger';
 import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import { getVersionById } from 'core/reducers/versions';
 import { sanitizeHTML } from 'core/utils';
+import { getClientCompatibility } from 'core/utils/compatibility';
 import Notice from 'ui/components/Notice';
+import type { AppState } from 'amo/store';
+import type { UserAgentInfoType } from 'core/reducers/api';
+import type { AddonVersionType } from 'core/reducers/versions';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
 
 import './style.scss';
 
-export class AddonCompatibilityErrorBase extends React.Component {
-  static propTypes = {
-    downloadUrl: PropTypes.string,
-    i18n: PropTypes.object.isRequired,
-    log: PropTypes.object,
-    minVersion: PropTypes.string.isRequired,
-    reason: PropTypes.string.isRequired,
-    userAgentInfo: PropTypes.object,
-  };
+type Props = {|
+  _getClientCompatibility?: typeof getClientCompatibility,
```

Comment:
If this is only used for dependency injection, you can probably move this line to the `InternalProps`.

Reply:
Based on our conversations about this issue on IRC, I felt like it would be better to be consistent about this, rather than sometimes use `InternalProps` (when a prop is only used for DI) and sometimes use `Props`. But I guess Flow will still protect us from misuse, and we can avoid `invariant`s by using `InternalProps`, so I'll change these.

Reply:
I don't want to add `invariant` statements for `_log`, `_config`, etc. So I'd like to give these rules of thumb:

- every prop that is part of the public API of the component should be added to `Props`
- every prop that is injected for testing purpose only (a few helper functions, `_log` and `_config` basically) should be declared as required in `InternalProps`